### PR TITLE
Chore: Refactor tempo using SDK contracts

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by zoltanbedi — APPROVED** [review]
LGTM, works nicely. 🌞

File: pkg/tsdb/tempo/tempo_test.go [hunk 1]
```diff
@@ -4,21 +4,15 @@ import (
...
-	require.NoError(t, err)
-
-	t.Run("createRequest should set Auth header when basic auth is true ", func(t *testing.T) {
-		req, err := executor.createRequest(context.Background(), &models.DataSource{BasicAuth: true, BasicAuthUser: "john", BasicAuthPassword: "pass"}, "traceID")
+	t.Run("createRequest - success", func(t *testing.T) {
+		service := &Service{}
+		req, err := service.createRequest(context.Background(), &datasourceInfo{}, "traceID")
 		require.NoError(t, err)
-		assert.Equal(t, 2, len(req.Header))
-		assert.NotEqual(t, req.Header.Get("Authorization"), "")
```
Comment (ying-jeanne) [inline pkg/tsdb/tempo/tempo_test.go:22]: Nice work! LGTM, except for this small thing that is not equal, I think we might want it here after changes?

Comment (idafurjes) [reply pkg/tsdb/tempo/tempo_test.go:22]: the authorization is not set anymore in this method, not sure how to test it

**Comment by ying-jeanne** [comment]
here is why we don't want simplejson anymore https://github.com/grafana/grafana/tree/main/pkg#avoid-use-of-the-simplejson-package, if you would like to clean it :)

**Comment by idafurjes** [comment]
I didn't know, sure I can do that on the other loki sdk PR

File: pkg/tsdb/tempo/tempo.go [hunk 2]
```diff
@@ -6,54 +6,102 @@ import (
...
 
-var (
-	tlog = log.New("tsdb.tempo")
-)
+func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	result := backend.NewQueryDataResponse()
+	queryRes := backend.DataResponse{}
+	refID := req.Queries[0].RefID
+
+	model, err := simplejson.NewJson(req.Queries[0].JSON)
```
Comment (ying-jeanne) [inline pkg/tsdb/tempo/tempo.go:86]: the simplejson is here.

**Pull request merged**